commit 13c9a9ded3ae744a1e11cbc14e9146d9fa427040
Author: Daniel Stenberg <daniel@haxx.se>
Date:   Sat Oct 7 00:11:31 2017 +0200

    imap: if a FETCH response has no size, don't call write callback
    
    CVE-2017-1000257
    
    Reported-by: Brian Carpenter and 0xd34db347
    Also detected by OSS-Fuzz: https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=3586

diff --git a/lib/imap.c b/lib/imap.c
index 954d18f37..baa31a2f8 100644
--- a/lib/imap.c
+++ b/lib/imap.c
@@ -1126,6 +1126,11 @@ static CURLcode imap_state_fetch_resp(struct connectdata *conn, int imapcode,
         /* The conversion from curl_off_t to size_t is always fine here */
         chunk = (size_t)size;
 
+      if(!chunk) {
+        /* no size, we're done with the data */
+        state(conn, IMAP_STOP);
+        return CURLE_OK;
+      }
       result = Curl_client_write(conn, CLIENTWRITE_BODY, pp->cache, chunk);
       if(result)
         return result;
